lightwood/api/json_ai.py

```diff
@@ -391,19 +392,20 @@ def generate_json_ai(
             ]
         )
         nr_mixers = len(list(outputs.values())[0].mixers)
-        encoder_time_budget_pct = max(
-            3.3 / 5, 1.5 + np.log(nr_trainable_encoders + 1) / 5
-        )
+        encoder_time_budget_pct = (2 + np.log(nr_trainable_encoders + 2)) / 5
+        if encoder_time_budget_pct > 4 / 5:
+            encoder_time_budget_pct = 4 / 5
 
         if nr_trainable_encoders == 0:
             problem_definition.seconds_per_encoder = 0
         else:
             problem_definition.seconds_per_encoder = int(
-                problem_definition.time_aim
+                (problem_definition.time_aim - problem_definition.expected_additional_time)
                 * (encoder_time_budget_pct / nr_trainable_encoders)
             )
         problem_definition.seconds_per_mixer = int(
-            problem_definition.time_aim * ((1 / encoder_time_budget_pct) / nr_mixers)
+            (problem_definition.time_aim - problem_definition.expected_additional_time)
```

Comment:
Is this saying that ensembling + analysis are not included in the time aim calculations? If so, add it to the docstring in `ProblemDefinition`.